File: rust/src/models/metric_summary.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    billable_metric_id::BillableMetricId, billing_metric_aggregate_enum::BillingMetricAggregateEnum,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MetricSummary {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub aggregation_key: Option<String>,

    pub aggregation_type: BillingMetricAggregateEnum,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub archived_at: Option<String>,

    pub code: String,

    pub created_at: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    pub id: BillableMetricId,

    pub name: String,
}

impl MetricSummary {
    pub fn new(
        aggregation_type: BillingMetricAggregateEnum,
        code: String,
        created_at: String,
        id: BillableMetricId,
        name: String,
    ) -> Self {
        Self {
            aggregation_key: None,
            aggregation_type,
            archived_at: None,
            code,
            created_at,
            description: None,
            id,
            name,
        }
    }
}

File: typescript/src/models/metricSummary.ts
// this file is @generated
import { type BillableMetricId, BillableMetricIdSerializer } from "./billableMetricId";
import {
  type BillingMetricAggregateEnum,
  BillingMetricAggregateEnumSerializer,
} from "./billingMetricAggregateEnum";

export interface MetricSummary {
  aggregationKey?: string | null;

  aggregationType: BillingMetricAggregateEnum;

  archivedAt?: Date | null;

  code: string;

  createdAt: Date;

  description?: string | null;

  id: BillableMetricId;

  name: string;
}

export const MetricSummarySerializer = {
  _fromJsonObject(object: any): MetricSummary {
    return {
      aggregationKey: object["aggregation_key"],
      aggregationType: BillingMetricAggregateEnumSerializer._fromJsonObject(
        object["aggregation_type"]
      ),
      archivedAt:
        object["archived_at"] != null ? new Date(object["archived_at"]) : undefined,
      code: object["code"],
      createdAt: new Date(object["created_at"]),
      description: object["description"],
      id: BillableMetricIdSerializer._fromJsonObject(object["id"]),
      name: object["name"],
    };
  },

  _toJsonObject(self: MetricSummary): any {
    return {
      aggregation_key: self.aggregationKey,
      aggregation_type: BillingMetricAggregateEnumSerializer._toJsonObject(
        self.aggregationType
      ),
      archived_at: self.archivedAt,
      code: self.code,
      created_at: self.createdAt,
      description: self.description,
      id: BillableMetricIdSerializer._toJsonObject(self.id),
      name: self.name,
    };
  },
};
